### ISSUE TITLE
Find where second-order is failing and create further issues to fix

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/math#403]
#### Summary:

@syclik claims that if we try to build all the models in `stan-dev/stan/test-models/good` for `fvar<var>` that they will break because second-order autodiff isn't complete.

#### Description:

Find where it's breaking.

#### Reproducible Steps:

Need to hack the test in `stan-dev/stan` to compile for second-order or do something else similar like create a new make target.

#### Current Output:

Failing?

#### Expected Output:

Everything buildable.

#### Current Version:

v2.12.0

**Comment by syclik** [comment]
It is failing. I need to figure out how to run through all the tests, but the first to fail: `asinh` can't be instantiated with `fvar<fvar<var>>`.

**Comment by bgoodri** [comment]
Should be able to pass the `-k` flag to make inside `runTests.py`.

**Comment by syclik** [comment]
I'm editing the first post to have a checklist of the failing instantiations.

**Comment by syclik** [comment]
For getting us to compile the Stan language with second-order autodiff, we're not as bad as I thought. We're missing some functions and the ODEs. The ODEs might take some effort to untangle. Since the introduction of CVODES, the code has gotten less general and more dependent on `stan::math::var`.

**Comment by bob-carpenter** [comment]
I think we should figure out a way to leave out ODEs to start
if we can.

**Comment by syclik** [comment]
Agreed. That's why I split out the list. Anyone with time could tackle the
first 3 functions.
